ql/src/java/org/apache/hadoop/hive/ql/parse/UpdateDeleteSemanticAnalyzer.java

```diff
@@ -271,6 +280,9 @@ private void reparseAndSuperAnalyze(ASTNode tree, Table mTable, ASTNode tabNameN
     rewrittenCtx.setEnableUnparse(false);
     analyzeRewrittenTree(rewrittenTree, rewrittenCtx);
 
+    // Check if metadata delete is possible and convert the tasks to metadata delete.
+    checkAndPerformStorageMetadataUpdate(mTable, storageHandler);
```

Comment:
why are we doing this after the query rewrite? that should be the same regardless of COW or MOR